qiskit_nature/second_q/mappers/bosonic_linear_mapper.py

```diff
@@ -108,7 +108,10 @@ def _map_single(
             bos_op_to_pauli_op = SparsePauliOp(["I" * qubit_register_length], coeffs=[1.0])
             for op, idx in terms:
                 if op not in ("+", "-"):
-                    break
+                    raise ValueError(
+                        f"Invalid bosonic operator: `{op}_{idx}`."
+                        + "All bosonic operators must have the following shape: `+_k` or `-_k`."
```

Comment:
```suggestion
                        "All bosonic operators must have the following shape: `+_k` or `-_k`."
```
We can rely on automatic string concatenation here :+1: